blockchains/bnb/lib/fetch_transactions.js

```diff
@@ -29,21 +43,35 @@ function sendTrxQuery(trxId, queue, metrics) {
   })
 }
 
-async function fetchTimeInterval(queue, startTimeMsec, endTimeMsec, nodeResponsePromises, metrics) {
+function getNumPagesToIterate(firstPageResult, startTimeMsec, endTimeMsec, bnbTradesMode) {
+  if (typeof firstPageResult === 'undefined' ||
+   (bnbTradesMode && firstPageResult.trade === 'undefined') ||
+   (!bnbTradesMode && firstPageResult.txArray === 'undefined')) {
+      throw ("Error in fetch interval ", startTimeMsec, " - ", endTimeMsec)
+  }
+
+  return bnbTradesMode ?
+  Math.ceil(firstPageResult.total / constants.NUM_TRADE_ROWS_FETCH)
```

Comment:
This additional parameter that we pass everywhere, oh, I wish we can avoid it. 
Can this switch be expressed with OOP somehow?
And this function in general? Why does it has so many arguments if it requires only one. The rest of params are only used to check preconditions, right? Can we check preconditions in another place? Current function signature confuses me a bit. It should "fetch time interval", but actually it just calculates the number of 'pages' in `firstPageResult`

Reply:
The function is stateless even though the arguments are indeed a lot. I am not sure OOP is simpler to grasp compared to procedural programming. A function that has no state and acts only upon arguments, even if they are more than 2 as per OOP rules is fine, no? I have broken it into a `check` part and main computation to simplify the logic.